### ISSUE TITLE
How to implement mutation provider classes

### ISSUE THREAD

**Issue by taimo42** [graphql-dotnet/graphql-dotnet#1521]
I would like to create the mutation fields outside the mutation class, so that I can use DI to get all classes which implements e.g. IMutationFieldsProvider and then call a method to create the fields.

Here is my mutation class
```
public class MyMutation : ObjectGraphType
{
    public MyMutation(IEnumerable<IMutationFieldsProvider> mutationFieldsProviders)
    {
        Name = "MyMutation";
        foreach (var mutationFieldsProvider in mutationFieldsProviders)
        {
            mutationFieldsProvider.CreateMutationFields(this);
        }
    }
}
```

And this is one of my mutation providers
```
public class MyMutationsProvider : IMutationFieldsProvider
{
    public void CreateMutationFields(ObjectGraphType myMutationObject)
    {
        myMutationObject.Field<MyPayloadType, MyPayload>()
                .Name("addSomthing")
                .ResolveAsync(async context =>
                {
                   // do something
                });
    }
}
```
When I thy this, I get the following error:
`MyMutation fields must be an object with field names as keys or a function which returns such an object.`

It is working when I call the Field method directly in the constructor of the MyMutation.

Is there a possibility to split up the mutations so that there are not all implemented in the constructor of the mutation class?

**Comment by sungam3r** [comment]
Did you try the latest preview version?

**Comment by taimo42** [comment]
Thank you for the fast reaction.

I was checking my code again and tried several things.
At the end it turned out, that there was an issue with the DI. I was not registering `MyMutationsProvider` in the right way. I managed to fix this. Now it is working as expected.

**So this was not an issue with GraphQl. Sorry for the inconvenience.**

I'm in general very happy with this library. But I'm also looking forward to the new features, because I'm still working with the latest stable release.

**Comment by sungam3r** [comment]
You may find that in fact the latest preview versions are even more "stable" than 2.4.0, also a bunch of useful features were implemented since 2.4.0 release. I myself use the latest previews in production for a long time.
